apollo-router/src/services/mod.rs

```diff
@@ -302,28 +303,31 @@ impl QueryPlannerRequest {
     pub fn new(
         query: String,
         operation_name: Option<String>,
-        query_plan_options: QueryPlanOptions,
         context: Context,
     ) -> QueryPlannerRequest {
         Self {
             query,
             operation_name,
-            query_plan_options,
+            query_plan_options: QueryPlanOptions::default(),
             context,
         }
     }
 }
 
 assert_impl_all!(QueryPlannerResponse: Send);
-/// [`Context`] and [`QueryPlan`] for the response..
+/// [`Context`] and [`QueryPlan`] for the response.
 pub struct QueryPlannerResponse {
     pub content: QueryPlannerContent,
     pub context: Context,
 }
 
+/// Query, QueryPlan and Introspection data in an opaque type.
+#[derive(Debug, Clone, From)]
+pub struct QueryPlannerContent(pub(crate) QueryPlannerContentInner);
```

Comment:
Why is there an inner struct instead of doing `QueryPlannerResponse { pub(crate) content: QueryPlannerContent, ...}` like for options?

Reply:
In that case the `new` constructor/builder also becomes `pub(crate)`. Which… is probably fine

Reply:
yes, it should not be created by plugin writers

Reply:
Done